doc/specs/#1571 - New Tab Menu Customization/#1571 - New Tab Menu Customization.md

```diff
@@ -108,7 +111,15 @@ There are five `type`s of objects in this menu:
     enabling all other profiles to also be accessible.
   - The "name" of these entries will simply be the name of the profile
   - The "icon" of these entries will simply be the profile's icon
-  - This won't include any profiles that have been included via `match` entries.
+  - This won't include any profiles that have been included via `matchProfile`
+    entries (below)
+* `"type": "matchProfile"`: Expands to all the profiles that match a given
+  regex, which haven't already been included. This lets the user easily specify
+  a whole collection of profiles for a folder, without needing to add them all
+  manually.
+  - `"name"`, `"commandline"` or `"source"`: These three properties are used to
+    filter the list of profiles, based on the matching property in the profile
+    itself. The value is a regex to string compare the profile's value with.

```

Comment:
Do we want to allow simple (partial) string matching in addition to regexes?

Reply:
Huh. https://github.com/microsoft/terminal/pull/12584#discussion_r968934688 raises a good point. I figured that regex would just always be easier, but I forgot that matching on the `.`'s would be a common scenario that would be arguably worse. 

I dunno, I'm on a real regex kick as of late, so I think I'm personally leaning towards regex always, but this is a point of contention that I'll let the team veto me on.

Reply:
Might as well just make "source" an array and match the whole string. This should work practically just as well as regexes would, since the amount of distinct "source"s is fairly limited (they aren't really auto-generated after all) and would allow easier diagnostics like "oh you specified the same source twice but they're supposed to be unique".

Reply:
Meh, I know that I'm using a weird personal edge case here but what about the following?

I've got a bunch of profiles with commandline's like `cmd.exe /k #work 15` that open up a workspace. I want to stick all those in a folder. So I add a `{ "type": "matchProfile", "commandline": "#work" }` entry

Reply:
I think we can make `source` an array, while doing regex/(partial) string match for `name` and `commandline`

Reply:
after realizing https://github.com/microsoft/terminal/pull/12584/files#r1009785219, I'm inclined to leave this as-is